sympy/solvers/solvers.py

```diff
@@ -1136,6 +1136,25 @@ def _has_piecewise(e):
         if _has_piecewise(fi):
             f[i] = piecewise_fold(fi)
 
+    # expand trig if it is a Mul (and it is not the only generator)
+    # or if it is a univariate Add -- this is a heuristic
+    # choice; some of the more reasonable results might be obtained
+    # by having a dedicated trig solver that handled post-solution
+    # simplification, e.g. trying to get solutions for args of
+    # functions to be written in terms of pi, etc...
+    trig_rep = {}
+    tsaw = 0
+    for fi in f:
+        for t in fi.atoms(TrigonometricFunction):
+            if not t.has_free(*symbols):
+                continue
+            tsaw += 1
+            if (t.args[0].is_Mul or len(t.free_symbols & set(symbols)) == 1) and t.args[0] not in trig_rep:
+                trig_rep[t] = t.expand(trig=True)
```

Comment:
Are there any limits on what this will expand?
```
In [3]: len(str(sin(1000*x).expand(trig=True)))
Out[3]: 160851
```

Reply:
It is now much more limited. And a case like that is added as a test with a note.